packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/state_var/views.rs

```diff
@@ -0,0 +1,325 @@
+use std::{
+    cell::{Ref, RefCell},
+    ops::Deref,
+    rc::Rc,
+};
+
+use super::{Freshness, QueryUpdateRequests, StateVarInner};
+
+/// A mutable view of the value of the state variable.
+/// It includes methods that allow one to view and change the variable.
+#[derive(Debug)]
+pub struct StateVarMutableView<T: Default + Clone> {
+    /// Structure containing the value of the variable its meta data.
+    /// Since inner is in an Rc<RefCell>, it is shared with other views and could be changed by them.
+    inner: Rc<RefCell<StateVarInner<T>>>,
+
+    /// A change counter that can be compared to the change counter of inner
+    /// in order to determine if the state variable has changed since last viewed.
+    change_counter_when_last_viewed: u32,
+}
+
+/// A read-only view of the value of the state variable.
```

Comment:
I'm confused by the use of the word `value`. Doesn't a view have more functionality than just seeing the value?

Reply:
I think the term `value` must have come from the definition of `StateVar<T>` struct with its `value` field. This docstring is definitely outdated.